sympy/calculus/codomain.py

```diff
@@ -0,0 +1,159 @@
+from sympy import S, sympify, diff, limit, oo
+from sympy.calculus.singularities import singularities
+from sympy.sets.sets import Interval, Intersection, FiniteSet, Union, Complement, Set, EmptySet
+from sympy.solvers.solveset import solveset_real
+
+
+def codomain(func, domain, *syms):
+    """ Finds the range of a real-valued function, for a real-domain
+
+    Parameters
+    ==========
+
+    func: Expr
+          The expression whose range is to be found
+    domain: Union of Sets
+          The real-domain for the variable involved in function
+    syms: Tuple of symbols
+          Symbol whose domain is given
+
+    Raises
+    ======
+
+    NotImplementedError
+          The algorithms to find the range of the given function are
+          not yet implemented.
+    ValueError
+          The input is not valid.
+    RuntimeError
+          It is a bug, please report to the github issue tracker.
+
+    Examples
+    ========
+
+    >>> from sympy import Symbol, S, Interval, Union, FiniteSet, codomain
+    >>> x = Symbol('x', real=True)
+    >>> codomain(x**2, Interval(-1, 1), x)
+    [0, 1]
+    >>> codomain(x/(x**2 - 4), Union(Interval(-1, 3), FiniteSet(5)), x)
+    (-oo, 1/3] U [3/5, oo)
+    >>> codomain(x**2/(x**2 - 4), S.Reals, x)
+    (-oo, 0] U (1, oo)
+    """
+
+    func = sympify(func)
+    if not isinstance(domain, Set):
+        raise ValueError('A Set must be given, not %s: %s' % (type(domain), domain))
+
+    if len(syms) == 0:
+        raise ValueError("A Symbol or a tuple of symbols must be given: not %s" % (syms,))
```

Comment:
Simply: `raise ValueError("A Symbol or a tuple of symbols must be given)` , Since `syms` is empty, no need to format it's type into the string.

This change `(syms) --  >(syms,)` was suggested for only line 54, while formatting a tuple into a string.